sub-packages/bionemo-scdl/tests/bionemo/scdl/index/test_row_feature_index.py

```diff
@@ -20,7 +20,53 @@
 import pandas as pd
 import pytest
 
-from bionemo.scdl.index.row_feature_index import RowFeatureIndex
+from bionemo.scdl.index.row_feature_index import RowFeatureIndex, are_dicts_equal
+
+
+@pytest.fixture
+def dict1():
+    return {"a": np.array([1, 2, 3]), "b": np.array([4, 5, 6])}
+
+
+@pytest.fixture
+def dict2():
+    return {"a": np.array([1, 2, 3]), "b": np.array([4, 5, 6])}
+
+
+@pytest.fixture
+def dict3():
+    return {"a": np.array([1, 2, 3]), "b": np.array([7, 8, 9])}
+
+
+@pytest.fixture
+def dict4():
+    return {"a": np.array([1, 2, 3]), "c": np.array([4, 5, 6])}
+
+
+@pytest.fixture
+def empty_dict():
+    return {}
+
+
+def test_equal_dicts(dict1, dict2):
+    assert are_dicts_equal(dict1, dict2) is True
+
+
+def test_unequal_values(dict1, dict3):
+    assert are_dicts_equal(dict1, dict3) is False
+
+
+def test_unequal_keys(dict1, dict4):
+    assert are_dicts_equal(dict1, dict4) is False
+
+
+def test_different_lengths(dict1):
+    smaller_dict = {"a": np.array([1, 2, 3])}
+    assert are_dicts_equal(dict1, smaller_dict) is False
+
+
+def test_empty_dicts(empty_dict):
+    assert are_dicts_equal(empty_dict, empty_dict) is True
```

Comment:
Don't use fixtures here, I know it reduces duplication but it also makes you have to search to find the dicts that are being compared. Just redefine the dicts each time and make each test take no fixture inupts